hack/patches/013-hpa-operator.patch

```diff
@@ -0,0 +1,227 @@
+ 		for _, override := range obj.GetSpec().GetWorkloadOverrides() {
+@@ -61,39 +50,14 @@ func HighAvailabilityTransform(obj base.KComponent, log *zap.SugaredLogger) mf.T
+ 		replicas := int64(*ha.Replicas)
+ 
+ 		// Transform deployments that support HA.
+-		if u.GetKind() == "Deployment" && !haUnSupported(obj).Has(u.GetName()) && !hasHorizontalPodAutoscaler(obj).Has(u.GetName()) {
++		if u.GetKind() == "Deployment" && !haUnSupported(u.GetName()) && !hasHorizontalPodAutoscaler(u.GetName()) {
+ 			if err := unstructured.SetNestedField(u.Object, replicas, "spec", "replicas"); err != nil {
+ 				return err
+ 			}
+ 		}
+ 
+ 		if u.GetKind() == "HorizontalPodAutoscaler" {
+-			min, _, err := unstructured.NestedInt64(u.Object, "spec", "minReplicas")
+-			if err != nil {
+-				return err
+-			}
+-			// Do nothing if the HPA ships with even more replicas out of the box.
+-			if min >= replicas {
+-				return nil
+-			}
+-
+-			if err := unstructured.SetNestedField(u.Object, replicas, "spec", "minReplicas"); err != nil {
+-				return err
+-			}
+-
+-			max, found, err := unstructured.NestedInt64(u.Object, "spec", "maxReplicas")
+-			if err != nil {
+-				return err
+-			}
+-
+-			// Do nothing if maxReplicas is not defined.
+-			if !found {
+-				return nil
+-			}
+-
+-			// Increase maxReplicas to the amount that we increased,
+-			// because we need to avoid minReplicas > maxReplicas happenning.
+-			if err := unstructured.SetNestedField(u.Object, max+(replicas-min), "spec", "maxReplicas"); err != nil {
++			if err := hpaTransform(u, replicas); err != nil {
+ 				return err
+ 			}
+ 		}
+diff --git a/vendor/knative.dev/operator/pkg/reconciler/common/hpa.go b/vendor/knative.dev/operator/pkg/reconciler/common/hpa.go
+new file mode 100644
+index 000000000..fba4a35cd
+--- /dev/null
++++ b/vendor/knative.dev/operator/pkg/reconciler/common/hpa.go
+@@ -0,0 +1,89 @@
++/*
++Copyright 2023 The Knative Authors
++
++Licensed under the Apache License, Version 2.0 (the "License");
++you may not use this file except in compliance with the License.
++You may obtain a copy of the License at
++
++    http://www.apache.org/licenses/LICENSE-2.0
++
++Unless required by applicable law or agreed to in writing, software
++distributed under the License is distributed on an "AS IS" BASIS,
++WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
++See the License for the specific language governing permissions and
++limitations under the License.
++*/
++
++package common
++
++import (
++	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
++	"k8s.io/apimachinery/pkg/util/sets"
++)
++
++// When a Podspecable has HPA, the replicas should be controlled by HPAs minReplicas instead of operator.
++// Hence, skip changing the spec.replicas for these Podspecables.
++func hasHorizontalPodAutoscaler(name string) bool {
++	return sets.NewString(
++		"webhook",
++		"activator",
++		"3scale-kourier-gateway",
++		"eventing-webhook",
++		"mt-broker-ingress",
```

Comment:
I don't think we're having HPA for mt-broker yet in eventing 1.10

(I don't see it in https://github.com/openshift-knative/serverless-operator/blob/main/openshift-knative-operator/cmd/operator/kodata/knative-eventing/1.10/4-mt-channel-broker.yaml 

It seems we are removing them by https://github.com/openshift-knative/eventing/blob/release-v1.10/openshift/release/generate-release.sh , although not sure if we actually still want to be doing that)

Reply:
opening a thread in https://redhat-internal.slack.com/archives/CEXRYS5QC/p1695999818126169

Reply:
opening https://issues.redhat.com/browse/SRVKE-1523  to re-add the HPAs into serverless 1.31

Reply:
Okay, we should wait for SRVKE-1523, then.